### PR TITLE
Add couple more selectors

### PR DESCRIPTION
**Pull request by PavelVanecek**
## Description

Moving things to redux store

## Related Issue

https://github.com/recharts/recharts/issues/4549

## Motivation and Context

I want to move the `getMouseInfo` function to Redux and there's some more information I need there, so this move.

## How Has This Been Tested?

npm test

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added a storybook story or extended an existing story to show my changes

### CODE REVIEW

File: test/context/chartLayoutContext.spec.tsx [hunk 1]
```diff
@@ -1325,3 +1327,87 @@ describe('ChartLayoutContextProvider', () => {
...
+    };
+    render(
+      <ComposedChart width={100} height={200} margin={{ top: 10, right: 20, bottom: 30, left: 40 }}>
+        <Customized component={Comp} />
+        <Brush height={13} />
+      </ComposedChart>,
+    );
+  });
+
+  it.todo('should include width of YAxis');
```
Comment (PavelVanecek) [inline test/context/chartLayoutContext.spec.tsx:1400]: We will need these if we want to move offset calculation to redux completely

**Comment by PavelVanecek** [comment]
Oh no. I will fix the type error tomorrow.

File: src/context/chartLayoutContext.tsx [hunk 2]
```diff
@@ -310,8 +308,10 @@ export const useViewBox = (): CartesianViewBox => {
   return useContext(ViewBoxContext);
 };
 
+const manyComponentsThrowErrorsIfOffsetIsUndefined: ChartOffset = {};
```
Comment (ckifer) [inline src/context/chartLayoutContext.tsx:311]: lol

File: src/state/layoutSlice.ts [hunk 3]
```diff
@@ -16,9 +28,15 @@ const chartLayoutSlice = createSlice({
     setLayout(state, action: PayloadAction<LayoutType>) {
       state.layoutType = action.payload;
     },
+    setContainer(state, action: PayloadAction<HasDomRect | null>) {
+      state.container = castDraft(action.payload);
```
Comment (ckifer) [inline src/state/layoutSlice.ts:32]: You dispatch an HTML element but only type one function?

File: src/chart/RechartsWrapper.tsx [hunk 4]
```diff
@@ -12,18 +14,24 @@ export type RechartsWrapperProps = {
...
-      {...wrapperEvents}
-      ref={ref}
-    >
-      {children}
-    </div>
-  ),
+  ({ children, width, height, className, style, wrapperEvents }: RechartsWrapperProps, ref: Ref<HTMLDivElement>) => {
+    const dispatch = useAppDispatch();
+    const innerRef = (node: HTMLDivElement | null) => {
+      dispatch(setContainer(node));
```
Comment (ckifer) [inline src/chart/RechartsWrapper.tsx:20]: Hmm, we probably shouldn't be dispatching an element into the store? But all good

**Review by ckifer — APPROVED** [review]

File: src/state/layoutSlice.ts (hunk 3 above)
Comment (ckifer) [reply src/state/layoutSlice.ts:32]: I guess that's to prevent us from doing stupid stuff. But the other properties will still be there

Comment (PavelVanecek) [reply src/state/layoutSlice.ts:32]: The interface is there to make mocks and unit testing easier, and to show which properties of the element is recharts using.

File: src/chart/RechartsWrapper.tsx (hunk 4 above)
Comment (PavelVanecek) [reply src/chart/RechartsWrapper.tsx:20]: Correct, Redux will consider this an antipattern. I tried to only dispatch the domRect, and sizes (which is the correct Redux way) but then the tooltip will not react to resize events. We can solve that with Resize observer... Later.

Comment (ckifer) [reply src/chart/RechartsWrapper.tsx:20]: Ack, makes sense

**Pull request merged**